sys/posix/sockets/posix_sockets.c

```diff
@@ -433,6 +432,9 @@ int bind(int socket, const struct sockaddr *address, socklen_t address_len)
             errno = EOPNOTSUPP;
             return -1;
     }
+    if (port.u16 != 0) {
```

Comment:
necessary? `byteorder_ntohs` will most likely use a built-in instruction with a low cycle count (1 for cortex platforms [REV16] e.g.).
Personally, I would rather take that cost instead of introducing a new branching.